File: Backend/Martins.Conciliacao.API/Filters/HeaderFilter.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Http.Filters
Imports System.Web.Http.Controllers
Imports System.Net
Imports System.Net.Http
Imports JsonWebToken.Model
Imports System.IdentityModel.Tokens
Imports Newtonsoft.Json
Imports JsonWebToken.Util
Imports Martins.Conciliacao.Util

Namespace Filters
  Public Class HeaderFilter
    Inherits ActionFilterAttribute

    Public Overrides Sub OnActionExecuting(ByVal actionContext As HttpActionContext)
      Dim request = actionContext.Request
      Dim headers = request.Headers

      If headers.Authorization Is Nothing OrElse String.IsNullOrEmpty(headers.Authorization.Parameter) Then
        Dim transactionInformation As TransactionalInformation = New TransactionalInformation()
        transactionInformation.ReturnMessage.Add("Invalid token")
        transactionInformation.ReturnStatus = False
        actionContext.Response = request.CreateResponse(Of TransactionalInformation)(HttpStatusCode.BadRequest, transactionInformation)
        Return
      End If

      Dim accessToken = headers.Authorization.Parameter
      Dim jwtToken = New JwtSecurityToken(accessToken)
      Dim userCode = jwtToken.Claims.FirstOrDefault(Function(claim) claim.Type = "ldap_usercode").Value
      Dim rolesJson = jwtToken.Claims.FirstOrDefault(Function(claim) claim.Type = "roles_json").Value
      Utils.UserCodeLDAP = userCode
      Utils.RolesObject = If(String.IsNullOrEmpty(rolesJson), New List(Of Object)(), JsonConvert.DeserializeObject(Of List(Of Object))(rolesJson))
    End Sub
  End Class
End Namespace

File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Filters/HeaderFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Http.Filters;
using System.Web.Http.Controllers;
using System.Net;
using System.Net.Http;
using JsonWebToken.Model;
using System.IdentityModel.Tokens;
using Newtonsoft.Json;
using JsonWebToken.Util;

namespace JsonWebToken.Filters
{
  public class HeaderFilter : ActionFilterAttribute
  {
    public override void OnActionExecuting(HttpActionContext actionContext)
    {
      var request = actionContext.Request;
      var headers = request.Headers;

      if (headers.Authorization == null || string.IsNullOrEmpty(headers.Authorization.Parameter))
      {
        TransactionalInformation transactionInformation = new TransactionalInformation();
        transactionInformation.ReturnMessage.Add("Invalid token");
        transactionInformation.ReturnStatus = false;

        actionContext.Response = request.CreateResponse<TransactionalInformation>(HttpStatusCode.BadRequest, transactionInformation);
        return;
      }

      var accessToken = headers.Authorization.Parameter;

      var jwtToken = new JwtSecurityToken(accessToken);

      // claims
      var userCode = jwtToken.Claims.FirstOrDefault(claim => claim.Type == "ldap_usercode").Value;
      var rolesJson = jwtToken.Claims.FirstOrDefault(claim => claim.Type == "roles_json").Value;

      Utils.UserCodeLDAP = userCode;
      Utils.RolesObject = string.IsNullOrEmpty(rolesJson) ? new List<object>() : JsonConvert.DeserializeObject<List<object>>(rolesJson);
    }
  }
}
